boa/src/builtins/object/internal_methods.rs

```diff
@@ -365,21 +365,24 @@ impl Object {
         self.properties.remove(name);
     }
 
-    /// Inserts a field in the object `properties` without checking if it's writable.
+    /// Inserts a field in the object `properties`, given an attribute without checking if it's writable.
     ///
     /// If a field was already in the object with the same name that a `Some` is returned
     /// with that field, otherwise None is retuned.
     #[inline]
-    pub(crate) fn insert_field<N>(&mut self, name: N, value: Value) -> Option<Property>
+    pub(crate) fn insert_property<N, V>(
+        &mut self,
+        name: N,
+        value: V,
+        attribute: Attribute,
+    ) -> Option<Property>
     where
-        N: Into<RcString>,
+        N: Into<PropertyKey>,
+        V: Into<Value>,
     {
         self.properties.insert(
-            name.into(),
-            Property::data_descriptor(
-                value,
-                Attribute::WRITABLE | Attribute::ENUMERABLE | Attribute::CONFIGURABLE,
-            ),
+            (&name.into()).into(),
```

Comment:
Why do we do this cant we just do:
```suggestion
            name.into(),
```

Reply:
`name.into()` changes it to `PropertyKey`. `(&name.into()).into()` changes it to `RcString`.

Reply:
Is there any way to get around that?

Reply:
Oh yeah. sorry forgot I fixed this in #603. this is a bug it converts a the `Symbol` into a string and stores in the properties.

Hmmm. In this leave it as is for now.